### ISSUE TITLE
Error starting Chrome: Message: unknown error: cannot connect to chrome -  cannot access local variable 'driver' - unhandled exception

### ISSUE THREAD

**Issue by slrslr** [FlareSolverr/FlareSolverr#1379]
### Have you checked our README?

- [X] I have checked the README

### Have you followed our Troubleshooting?

- [X] I have followed your Troubleshooting

### Is there already an issue for your problem?

- [X] I have checked older issues, open and closed

### Have you checked the discussions?

- [X] I have read the Discussions

### Have you ACTUALLY checked all these?

YES

### Environment

```markdown
- FlareSolverr version: 3.3.21
- Last working FlareSolverr version: -
- Operating system: Debian 12
- Are you using Docker: [no]
- FlareSolverr User-Agent (see log traces or / endpoint):
- Are you using a VPN: [yes]
- Are you using a Proxy: [no]
- Are you using Captcha Solver: [no]
```

### Description

```
wget https://github.com/FlareSolverr/FlareSolverr/releases/download/v3.3.21/flaresolverr_linux_x64.tar.gz --show-progress; tar -xzf flaresolverr_linux_x64.tar.gz; cd flaresolverr && ./flaresolverr
```

output:

> 2024-10-02 09:37:47 INFO     FlareSolverr 3.3.21
> 2024-10-02 09:37:47 INFO     Testing web browser installation...
> 2024-10-02 09:37:47 INFO     Platform: Linux-5.10.0-32-amd64-x86_64-with-glibc2.36
> 2024-10-02 09:37:47 INFO     Chrome / Chromium path: /dev/shm/flaresolverr/chrome/chrome
> 2024-10-02 09:37:47 INFO     Chrome / Chromium major version: 123
> 2024-10-02 09:37:47 INFO     Launching web browser...
> 2024-10-02 09:38:53 ERROR    Error starting Chrome: Message: unknown error: cannot connect to chrome at 127.0.0.1:35437
> from chrome not reachable
> Stacktrace:
> #0 0x55b9ccd81863 <unknown>
> #1 0x55b9cca77717 <unknown>
> #2 0x55b9cca627c3 <unknown>
> #3 0x55b9ccab0c91 <unknown>
> #4 0x55b9ccaa77bb <unknown>
> #5 0x55b9ccaf0aed <unknown>
> #6 0x55b9ccae4343 <unknown>
> #7 0x55b9ccab5593 <unknown>
> #8 0x55b9ccab5f5e <unknown>
> #9 0x55b9ccd4584b <unknown>
> #10 0x55b9ccd497a5 <unknown>
> #11 0x55b9ccd33571 <unknown>
> #12 0x55b9ccd4a332 <unknown>
> #13 0x55b9ccd1887f <unknown>
> #14 0x55b9ccd70728 <unknown>
> #15 0x55b9ccd708fb <unknown>
> #16 0x55b9ccd809b4 <unknown>
> #17 0x7f527fe76144 <unknown>
> 
> Traceback (most recent call last):
>   File "utils.py", line 316, in get_user_agent
>   File "utils.py", line 203, in get_webdriver
> UnboundLocalError: cannot access local variable 'driver' where it is not associated with a value
> 
> During handling of the above exception, another exception occurred:
> 
> Traceback (most recent call last):
>   File "flaresolverr.py", line 108, in <module>
>   File "flaresolverr_service.py", line 73, in test_browser_installation
>   File "utils.py", line 322, in get_user_agent
> Exception: Error getting browser User-Agent. cannot access local variable 'driver' where it is not associated with a value
> [2310934] Failed to execute script 'flaresolverr' due to unhandled exception!

I am using a working VPN. The server has firewall OUTPUT policy ACCEPT and INPUT policy DROP.
I am trying to run FlareSolverr in order to get "FlareSolverr API URL:" to input into Jackett http://127.0.0.1:9117/UI/Dashboard which I hope will allow connection to the Jackett indexers/trackers.

Thank you and it may be handy if the exception is handled more user friendly so issue like mine does not need to be filled.

### Logged Error Messages

```text
no *log* in flaresolverr directory downloaded
```

**Comment by ilike2burnthing** [comment]
Have a look at this issue - https://github.com/FlareSolverr/FlareSolverr/issues/1115

Try setting the `HOST = localhost` ENV, then try disabling the VPN. Failing those, you could try running https://github.com/21hsmw/FlareSolverr/tree/nodriver-support from source.

**Comment by slrslr** [comment]
Why would VPN be a problem if all OS apps works without any special configuration?
I have tried "HOST=localhost" command and re-run Flaresolverr with same result:
```
ERROR    Error starting Chrome: Message: unknown error: cannot connect to chrome at 127.0.0.1:45457
from chrome not reachable
```
Unfortunately I am not willing to try running without a VPN protection. You have said to try nodriver-support from source. I can not use pre-compiled binary flaresolverr_linux_x64.tar.gz from https://github.com/FlareSolverr/FlareSolverr/releases or i would end up with same result likely (turning off the VPN is not an option for me). Thank you

**Comment by elazar** [comment]
I seem to be running into a similar issue.

```
- FlareSolverr version: 3.3.21
- Operating system: Debian 11
- Are you using Docker: [yes]
- FlareSolverr User-Agent (see log traces or / endpoint): see attached
- Are you using a VPN: [no]
- Are you using a Proxy: [no]
- Are you using Captcha Solver: [no]
```

[flaresolverr.log](https://github.com/user-attachments/files/17310563/flaresolverr.log)

**Comment by ilike2burnthing** [comment]
Use `21hsmw/flaresolverr:nodriver`.

**Comment by elazar** [comment]
@ilike2burnthing I'm afraid that doesn't resolve the issue. I tried running both the `nodriver` and `nodrivertest` (as referenced in #1163) tags of the image you referenced with `DRIVER` set to `nodriver` and `LOG_LEVEL` set to `debug`. Both still encounter the referenced error message. I've attached logs of the output for each.

[flaresolverr-nodriver.log](https://github.com/user-attachments/files/17348152/flaresolverr-nodriver.log)

[flaresolverr-nodrivertest.log](https://github.com/user-attachments/files/17348154/flaresolverr-nodrivertest.log)
